cli: autocomplete: complete `--config` with possible configuration keys

jakobhellermann opened the pull request:
![image](https://github.com/user-attachments/assets/2ac6a34e-806c-4b2d-95b0-e2ed3233d1d4)


If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

yuja commented on CHANGELOG.md:
```diff
@@ -45,6 +45,8 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 * `jj util mangen` is replaced with `jj util install-man-pages`, which can
   install man pages for all `jj` subcommands to a given path.
 
+* Add dynamic autocompletion for the `jj --config key=value` argument
```
nit: omit the changelog entry? I don't think changelog entry is needed for improvement of completion. User doesn't have to know that beforehand.

yuja commented on cli/src/complete.rs:
```diff
@@ -409,20 +410,23 @@ fn config_keys_rec(
 }
 
-fn config_keys_impl(only_leaves: bool) -> Vec<CompletionCandidate> {
+fn config_keys_impl(only_leaves: bool, include_eq: bool) -> Vec<CompletionCandidate> {
```
nit: (at your option) `suffix: &str` could be added instead (to avoid `(bool, bool)`), but that's not important. One minor good thing is that we can use `format!("{name}{suffix}")` to construct an escaped key.

jakobhellermann replied:
good idea

yuja reviewed (approved):
Thanks.

jakobhellermann commented:
I've addressed the comments and added a test. So I can just go ahead and enable the automerge, right? Since the PR has been approved and the changes afterwards were minor (test and clippy fix).
